examples/provider/bid/conversant/conversant-ex1.html

```diff
@@ -0,0 +1,379 @@
+        };
+        xmlHttp.open('POST', url, true); // true for asynchronous
+        xmlHttp.withCredentials = true;
+        xmlHttp.send(data);
+      };
+
+      window.googletag = window.googletag || {};
+      googletag.cmd = googletag.cmd || [];
+      var pf = new pubfood();
+      pf.omitDefaultBidKey(true);
+      var batch = [];
+
+      // this reporter listens on all events and pushes the event data to a queue `batch`
+      pf.observe(function(event) {
+        batch.push(event);
+      });
+
+      // this reporter only listens on the `AUCTION_COMPLETE` event, then loops over and displays the event data
+      pf.observe('AUCTION_COMPLETE', function(event) {
+        console.log('---------------------------------------');
+        for (var i = 0; i < batch.length; i++) {
+          console.log('%c' + batch[i].type, 'padding-right:5px;font-weight:bold;display:block;', JSON.stringify(batch[i]));
+        }
+      });
+
+      var slot = pf.addSlot({
+        name: '/2476204/cn-multi-size',
+        sizes: [
+          [300, 250],
+          [300, 600]
+        ],
+        elementId: 'div-multi-size',
+        bidProviders: [
+          'conversant'
+        ]
+      });
+      slot.setParam('conversant', {
+        slot: 'medrec'
+      });
+
+      slot = pf.addSlot({
+        name: '/2476204/cn-leaderboard',
+        sizes: [
+          [728, 90]
+        ],
+        elementId: 'div-leaderboard',
+        bidProviders: [
+          'conversant'
+        ]
+      });
+      slot.setParam('conversant', {
+        slot: 'leaderboard'
+      });
+
+      var conversantProvider = pf.addBidProvider({
+        name: 'conversant',
+        init: function(slots, pushBid, done) {
+          conversantProvider.requestBids(slots, pushBid, done);
+        },
+        refresh: function(slots, pushBid, done) {
+          conversantProvider.requestBids(slots, pushBid, done);
+        }
+      });
+      conversantProvider.bids = [];
+      conversantProvider.bidderPrefix = 'cn';
+      conversantProvider.lastId = 1;
+
+      conversantProvider._getDNT = function() {
+        return navigator.doNotTrack === '1' || window.doNotTrack === '1' || navigator.msDoNotTrack === '1' || window.doNotTrack === 'yes';
+      };
+
+      conversantProvider._generateImpId = function(slotName , size) {
+        return slotName + '_' + size[0] + 'x' + size[1];
+      };
+
+      conversantProvider._parseImpId = function(impid) {
+        return {
+          slotName: impid.substring(0, impid.lastIndexOf('_')),
+          size: impid.substring(impid.lastIndexOf('_') + 1, impid.length).split('x')
+        };
+      };
+
+      conversantProvider._getNextUniqueId = function() {
+        conversantProvider.lastId += 1;
+        return conversantProvider.lastId;
+      };
+
+      conversantProvider._callbackFactory = function(slots,pushBid,done){
+        return function(rsp) {
+          rsp.seatbid.forEach(function(seatBid) {
```

Comment:
`Array.prototype.forEach` also assumes some browser ECMA JavaScript engine compliance.

In general, Pubfood needs to take a position on this. Generally later ECMA standards are "generally" accepted, but as a JS library provider we cannot always guarantee that, for example, a user agent will not be IE<9.